lib/parallel_tests/cucumber/scenario_line_logger.rb

```diff
@@ -11,20 +11,30 @@ def initialize(tag_expression = ::Cucumber::Core::Gherkin::TagExpression.new([])
           @tag_expression = tag_expression
         end
 
-        def visit_feature_element(uri, feature_element, feature_tags)
+        def visit_feature_element(uri, feature_element, feature_tags, line_numbers: [])
           scenario_tags = feature_element[:tags].map {|tag| ::Cucumber::Core::Ast::Tag.new(tag[:location], tag[:name])}
           scenario_tags = feature_tags + scenario_tags
           if feature_element[:examples].nil? # :Scenario
+            test_line = feature_element[:location][:line]
+
             # We don't accept the feature_element if the current tags are not valid
             return unless @tag_expression.evaluate(scenario_tags)
+            # or if it is not at the correct location
+            return unless line_numbers.empty? || line_numbers.include?(test_line)
```

Comment:
maybe easier to read with `return if line_numbers.any? && !line_numbers.include?(test_line)`

Reply:
The current way keeps the parallelism of phrasing (i.e. `rerun unless` for both conditions) and I suspect that community convention would prefer using `unless` rather than `if` plus negation but, yeah, I can change it to the other way.